flutter/flutter-provider/state-management-in-flutter-using-provider-Part2.md

```diff
@@ -0,0 +1,209 @@
+# State Management in Flutter using Provider: Part 2
+
+In previous [article](https://suresh.hashnode.dev/state-management-in-flutter-using-provider-part-1-cke8vk1cb00iny0s11lu933ch)
+we have discussed the problem of using `setState` to update the app's state in the moderate or complex app.
+
+## What we are going to know in this article
+
+1. Why we should use provider
+2. What are the types of provider
+3. How to use provider in our code
+
+In this article we'll know about `provider` package owned by  [Remi Rousselet](https://github.com/rrousselGit/provider), this is one of the best state management technique we can use in our app.
+
+### Why we should use provider
+
+ On the beginner stage, we generally use the `setState` to update the state of our app, because its easy to do so, and on this stage, if we try to use some moderate or say a complex type of state management technique like `flutter_bloc` or `Mobx`, it is not easy for beginners to take a big jump from using `setState` to these state management technique 😓
+
+ 
+``` 
+provider uses InheritedWidget internally, to manage all those complex computations and provide us simple and great feature to manage our state
+```
+
+&nbsp;
+
+> Why `provider`? We have some other techniques that can pass data in the widget tree.
+
+Yes, there is some other way, such as passing data via constructors this can be done, but definitely not the good approach.
```

Comment:
Can you explain why it's not a good approach?  
Also "Yes, there is some other way, such as passing data via constructors this can be done," is just saying the question again, you can directly start with "This is not considered a good approach/coding practice because..."